src/deliver/render_message.rs

```diff
@@ -196,17 +212,32 @@ fn truncate(s: &str, max_chars: usize) -> String {
     result.trim().to_string()
 }
 
-fn remove_html(string_with_maybe_html: &str) -> String {
-    let string_without_html = nanohtml2text::html2text(string_with_maybe_html);
+fn remove_html(string_with_maybe_html: &str, html_parser: &HtmlParser) -> String {
+    let string_without_html = match html_parser {
+        HtmlParser::Nano => nanohtml2text::html2text(string_with_maybe_html),
+        HtmlParser::Ammonia => ammonia::Builder::empty()
+            .link_rel(None)
+            .url_relative(ammonia::UrlRelative::Deny)
+            .clean(string_with_maybe_html)
+            .to_string(),
+    };
 
     let ac = AhoCorasickBuilder::new()
         .match_kind(MatchKind::LeftmostFirst)
         .build(&[
             "&#32;", "&", "<", ">", "\u{200B}", "\u{200C}", "\u{200D}", "\u{2060}", "\u{FEFF}",
         ]);
 
-    ac.replace_all(
+    let cleaned = ac.replace_all(
         &string_without_html,
         &[" ", "&amp;", "&lt;", "&gt;", " ", " ", " ", " ", " "],
-    )
+    );
+
+    match html_parser {
+        HtmlParser::Nano => cleaned,
+        HtmlParser::Ammonia => {
+            let regexp = regex::Regex::new(r"(\n(\s)*(\n)(\s)*\n)").unwrap();
```

Comment:
What is the reason for this unwrap ? 

Can be managed with `?` Operator ?

Reply:
because it's a hard-coded regexp and we're sure it's a valid one

Reply:
Nice then